From 01dab88ce354c9a565ec1ed192b79c7d219eb900 Mon Sep 17 00:00:00 2001
From: Tomas Popela <tpopela@redhat.com>
Date: Mon, 23 Mar 2015 13:18:39 +0100
Subject: [PATCH 0142/1800] Bug 740475 - Quoted reply editing issues

Check if won't add two BR elements in row to avoid unnecessary empty line.
---
 web-extensions/e-html-editor-view-dom-functions.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/web-extensions/e-html-editor-view-dom-functions.c b/web-extensions/e-html-editor-view-dom-functions.c
index b297db5b50..2bc68bf0af 100644
--- a/web-extensions/e-html-editor-view-dom-functions.c
+++ b/web-extensions/e-html-editor-view-dom-functions.c
@@ -1948,8 +1948,12 @@ body_keyup_event_cb (WebKitDOMElement *element,
 					WEBKIT_DOM_NODE (selection_start_marker)),
 				NULL);
 
-			webkit_dom_node_append_child (
-				node, WEBKIT_DOM_NODE (br_element), NULL);
+			if (!WEBKIT_DOM_IS_HTMLBR_ELEMENT (webkit_dom_node_get_last_child (node)))
+				webkit_dom_node_append_child (
+					node, WEBKIT_DOM_NODE (br_element), NULL);
+			else
+				remove_node (WEBKIT_DOM_NODE (br_element));
+
 			remove_node (parent);
 		}
  restore:
-- 
2.15.0

